prereise/gather/demanddata/nrel_efs/get_efs_data.py

```diff
@@ -36,129 +33,273 @@ def download_data(es=None, ta=None, fpath=""):
+    open(zip_name, "wb").write(r.content)
+    print(zip_name + " successfully downloaded!")
+
+    # Return the data to try extracting with Python's zipfile module
+    return zipfile.ZipFile(io.BytesIO(r.content))
+
+
+def _extract_data(z, zf_works, zip_name, csv_name, fpath):
+    """Extracts the .csv file containing NREL EFS data from the downloaded .zip file.
+    First attempts extraction using Python's zipfile module, then attempts other
+    OS-dependent methods, as needed.
+
+    :param zipfile.ZipFile z: The .zip file stored in memory for attempted extraction
+        using Python's zipfile module.
+    :param bool zf_works: An indicator flag that states whether or not Python's zipfile
+        module works for extraction. True if Python's zipfile module works, else False.
+    :param str zip_name: The name of the specified .zip file.
+    :param str csv_name: The name of the .csv file contained within the .zip file.
+    :param str fpath: The input file path.
+    :return: (*bool*) -- The indicator flag that states whether or not Python's zipfile
+        module works for extraction. This is returned to prevent checking Python's
+        zipfile module if it does not work the first time (in the event multiple .zip
+        files require extraction).
+    :raises NotImplementedError: if Python's zipfile module cannot extract the .csv
+        file.
+    :raises OSError: if an OS other than Windows, macOS, or Linux is identified.
+    """
 
+    # Assign the path name of the .zip file
+    zip_path = os.path.join(fpath, zip_name)
+
+    try:
+        if zf_works:
+            # Try the zipfile module first
+            z.extractall(fpath)
+            print(csv_name + " successfully extracted!")
+        else:
+            # Bypass the zipfile module if it does not work on the first file
+            raise NotImplementedError
+    except NotImplementedError:
+        if zf_works:
+            print(
+                zip_name
+                + " is compressed using a method that is not supported by the"
+                + " zipfile module."
+            )
+            print("Trying other extraction methods supported by your OS.")
+            zf_works = False
+
+        # Try other extraction methods depending on operating system
+        if platform.system() == "Windows":
             try:
-                if zf_works:
-                    # Try the zipfile module first
-                    z[i][j].extractall(fpath)
-                    print(csv_name + " successfully extracted!")
-                else:
-                    # Bypass the zipfile module if it does not work on the first file
-                    raise NotImplementedError
-            except NotImplementedError:
-                if zf_works:
+                # Windows Command Line does not support this type of compression
+                # Try using 7zip, if it is installed in the default location
+                sz_path = "C:/Program Files/7-Zip/7z.exe"
```

Comment:
Should the user be able to specify a path to an alternate location at which they've installed 7zip?

Reply:
Yeah I can definitely add that. I think I originally left it out because I wasn't sure how often people install it outside of the default location and if it was worth it to include an additional input. I can make it so the current location is the default input for an additional `sz_path` input.

Reply:
Done.